src/overlays/actors/ovl_En_Guard_Nuts/z_en_guard_nuts.c

```diff
@@ -28,42 +36,338 @@ const ActorInit En_Guard_Nuts_InitVars = {
+    this->actor.targetMode = 1;
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    Actor_SetScale(&this->actor, 0.015f);
+    Math_Vec3f_Copy(&this->unk228, &this->actor.world.pos);
+    this->unk21E = D_80ABBE1C;
+    D_80ABBE1C++;
+
+    // If you have returned deku princess guards will init burrowed.
+    if (!(gSaveContext.weekEventReg[23] & 0x20)) {
+        EnGuardNuts_SetupWait(this);
+    } else {
+        EnGuardNuts_Burrow(this, globalCtx);
+    }
+}
+
+void EnGuardNuts_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnGuardNuts* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+/**
+ * @brief Changes the animation to the provided index. Updates animIndex and animFrameCount for the animation.
+ *
+ * @param this
+ * @param index the index of sAnimations to change to
+ */
+void EnGuardNuts_ChangeAnim(EnGuardNuts* this, s32 index) {
+    this->animIndex = index;
+    this->animFrameCount = Animation_GetLastFrame(sAnimations[this->animIndex]);
+    Animation_Change(&this->skelAnime, sAnimations[this->animIndex], 1.0f, 0.0f, this->animFrameCount,
+                     sAnimModes[this->animIndex], -2.0f);
+}
+
+void EnGuardNuts_SetupWait(EnGuardNuts* this) {
+    EnGuardNuts_ChangeAnim(this, WAIT_ANIM);
+    this->unk21C = 0;
+    this->actionFunc = EnGuardNuts_Wait;
+}
+
+void EnGuardNuts_Wait(EnGuardNuts* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    s16 phi_a1;
+    s16 yawDiff;
+
+    SkelAnime_Update(&this->skelAnime);
+    yawDiff = ABS_ALT(BINANG_SUB(this->actor.yawTowardsPlayer, this->actor.home.rot.y));
+    if (Actor_ProcessTalkRequest(&this->actor, &globalCtx->state)) {
+        func_80ABB540(this);
+        return;
+    }
+    if (yawDiff > 0x6000) {
+        D_80ABBE20 = 2;
+    }
+    if (player->transformation == PLAYER_FORM_DEKU) {
+        // this is the palace of...
+        this->textId = 0;
+        if ((gSaveContext.weekEventReg[17] & 4) && (!this->hasCompletedConversation)) {
+            // I told you not to enter!!
+            this->textId = 7;
+        } else if (gSaveContext.weekEventReg[12] & 0x40) {
+            // come to see the monkey again?
+            this->textId = 4;
+        }
+    } else {
+        // No place for an outsider
+        this->textId = 3;
+    }
+    this->actor.textId = sTextIDs[this->textId];
+    phi_a1 = this->actor.world.rot.y;
+    if (D_80ABBE20 == 2) {
+        EnGuardNuts_Burrow(this, globalCtx);
+        return;
+    }
+    if (D_80ABBE20 == 1) {
+        if (this->animIndex != WALK_ANIM) {
+            EnGuardNuts_ChangeAnim(this, WALK_ANIM);
+        }
+        phi_a1 = (this->actor.home.rot.y + 0x8000);
+    }
+    if (fabsf((f32)(this->actor.shape.rot.y - phi_a1)) < 100.0f) {
+        this->actor.shape.rot.y = phi_a1;
+        if ((D_80ABBE20 == 1) && (this->animIndex != WALK_ANIM_2)) {
+            EnGuardNuts_ChangeAnim(this, WALK_ANIM_2);
```

Comment:
I don't... recall the guards at the front of the palace ever walking anywhere, are they reusing the walk animation without moving for something (surprise? head shaking? anger?) is this unused code? assuming the animation gets used in look nuts so we know its the real walking animation.

Reply:
When you finish talking with them, they rotate 180 degrees "walking" in place, that is where the walking animation comes in to play.